loading.bs

```diff
@@ -331,6 +390,433 @@ Note: Applying the signed exchange's response here has the effect of letting a
+
+The {{Document}} has a map of <dfn dfn>prefetched signed exchanges for
+navigation</dfn>, which is a [=map=] of [=URL=] to [=exchange=], initially
+empty.  <span class="note">This might merge with the set of prefetched resources
+when <{link/rel/prefetch}> is specified more completely.</span>
+
+The {{Document}} has a set of <dfn dfn>prefetched
+subresource signed exchanges</dfn>, which is a set of [=exchanges=], initially
+empty.
+
+</blockquote>
+
+<h3 algorithm id="mp-navigation-params">New navigation params struct field</h3>
+
+In the [=navigation params=], add the following items:
+
+<dl dfn-for="navigation params">
+
+: <dfn>prefetched subresource signed exchanges</dfn>
+:: A set of [=exchanges=], initially empty.
+
+</dl>
+
+<h3 algorithm id="mp-link-type-prefetch">
+Monkeypatch Link type "prefetch"</h3>
+
+The [=process the linked resource=] given a link element |el|, boolean
+|success|, and [=response=] |response|:
+
+1. If |success| is true, [=fire an event=] named {{HTMLElement/load}} at |el|.
+1. Otherwise, [=fire an event=] named {{HTMLElement/error}} at |el|.
+1. If the |success| is false, then return.
+1. If the |response|'s [=response/came from a signed exchange=] is false, then
+    return.
+1. Let |clonedExchange| be the result of creating a new [=exchange=] with the
+    first item of |response|'s [=response/URL list=] and the result of
+    [=response/cloning=] |response|.
+1. [=map/Set=] |el|'s [=Node/node document=]'s [=Document/prefetched signed
+    exchanges for navigation=][|clonedExchange|'s [=exchange/request URL=]] to
+    |clonedExchange|.
+1. Let |outerLinkHeader| be the result of [=header list/getting=] `` `Link` ``
+    from |response|'s [=response/signed exchange outer header list=].
+1. If |outerLinkHeader| is null, then return.
+1. Let |outerLinks| be the result of [=Parsing a Link Field Value=] from
+    |outerLinkHeader|.
+1. If |outerLinks| is empty, then return.
+1. Let |innerLinkHeader| be the result of [=header list/getting=] `` `Link` ``
+    from |response|'s [=response/header list=].
+1. If |innerLinkHeader| is null, then return.
+1. Let |innerLinks| be the result of [=Parsing a Link Field Value=] from
+    |innerLinkHeader|.
+1. If |innerLinks| is empty, then return.
+1. Let |alternateLinks| be an empty [=list=].
+1. For each |outerLink| of |outerLinks|:
+    1. If |outerLink|'s [=Relation Type=] is not 'alternate', then continue.
+    1. If |outerLink|'s [=Target Attributes list|Target Attributes=] doesn't
+        [=list/contain=] (`"type"`,`"application/signed-exchange"`), then
+        continue.
+    1. Let |linkTarget| be |outerLink|'s [=Link Target=].
+    1. Let |linkContext| be |outerLink|'s [=Link Context=].
+    1. Let |outerLinkVariants| be the second item of the first tuple of
+        |outerLink|'s [=Target Attributes list|Target Attributes=] whose first
+        item matches the string `"variants"` or the empty string ("") if it is
+        not present.
```

Comment:
I would probably define a small algorithm for this bit to make it more readable, since it's used so many times. E.g.

> `A link header value |link|'s <dfn noexport>Target Attribute named</dfn> |name| is the second item of the first tuple of |link|'s [=Target Attributes list|Target Attributes=] whose first item matches the string |name|, or the empty string ("") if no such tuple is present.`

And then use it like
```
Let |outerLinkVariants| be |outerLink|'s [=Target Attribute named=] `"variants"`.
```